recipes/dirent/all/conanfile.py

```diff
@@ -0,0 +1,32 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class DirEntConan(ConanFile):
+    name = "dirent"
+    description = "Dirent is a C/C++ programming interface that allows programmers to retrieve information about " \
+                  "files and directories under Linux/UNIX"
+    topics = ("conan", "dirent", "directory", "file system")
+    homepage = "https://github.com/tronkko/dirent"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "compiler"
+    no_copy_source = True
+
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        if self.settings.compiler != "Visual Studio":
```

Comment:
```suggestion
        if self.settings.os != "Windows" or self.settings.compiler != "gcc":
```

Reply:
My initial suggestion was bad because `"Visual Studio" != "gcc"`

I propose the following:
```
    def configure(self):
        if self.settings.os != "Windows":
            raise ConanInvalidConfiguration("Only Windows builds are supported")
        if self.settings.compiler == "gcc":
            raise ConanInvalidConfiguration("mingw has a dirent.h implementation")

```

Reply:
It looks better! Please, apply it.